api/machinemanager/machinemanager.go

```diff
@@ -49,6 +49,48 @@ func (client *Client) ForceDestroyMachines(machines ...string) ([]params.Destroy
 	return client.destroyMachines("ForceDestroyMachine", machines)
 }
 
+// DestroyMachinesWithParams removes the given set of machines, the semantics of which
+// is determined by the force and keep parameters.
+// TODO(wallyworld) - for Juju 3.0, this should be the preferred api to use.
+func (client *Client) DestroyMachinesWithParams(force, keep bool, machines ...string) ([]params.DestroyMachineResult, error) {
+	if !keep {
+		if force {
+			return client.ForceDestroyMachines(machines...)
+		}
+		return client.DestroyMachines(machines...)
+	}
+	args := params.DestroyMachinesParams{
+		Force:       force,
+		Keep:        keep,
+		MachineTags: make([]string, 0, len(machines)),
+	}
+	allResults := make([]params.DestroyMachineResult, len(machines))
```

Comment:
Given that the API server needs to do these checks anyway, why duplicate them and make the call site more complicated, on the client side?